src/vivarium/framework/state_machine.py

```diff
@@ -406,7 +406,7 @@ def setup(self, builder: Builder) -> None:
             Interface to several simulation tools including access to common random
             number generation, in particular.
         """
-        self.random = builder.randomness.get_stream(self.name)
+        self.random = builder.randomness.get_stream(self.name, self)
```

Comment:
Nit: I'd much prefer self.randomness for all random stream names unless a component needs multiple.